main.go

```diff
@@ -115,8 +138,60 @@ func main() {
 	}
 
 	setupLog.Info("starting manager")
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+	if err := mgr.Start(ctx); err != nil {
 		setupLog.Error(err, "problem running manager")
 		os.Exit(1)
 	}
 }
+
+// Tee consumes the received channel and mirrors the messages into 2 new channels.
+func tee[T any](ctx context.Context, in <-chan T) (_, _ <-chan T) {
+	out1, out2 := make(chan T), make(chan T)
+
+	OrDone := func(ctx context.Context, in <-chan T) <-chan T {
+		out := make(chan T)
+		go func() {
+			defer close(out)
+
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case i, ok := <-in:
+					if !ok {
+						return
+					}
+					select {
+					case out <- i:
+					case <-ctx.Done():
+					}
+				}
+			}
+		}()
+		return out
+	}
+
+	go func() {
+		defer close(out1)
+		defer close(out2)
+
+		for val := range OrDone(ctx, in) {
+			select {
+			case <-ctx.Done():
+
```

Comment:
Are we supposed to return here?

Reply:
Yes. We will return because the channel returned from `OrDone` is going to be closed, hence the loop at 178 will end.
I probably should add a return statement at line 181 for clarity though